tests/virt_autotest/switch_to_ssh_and_install_hypervisor.pm

```diff
@@ -0,0 +1,56 @@
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: For openSUSE virtualization test only. login in console, install kvm/xen patterns if needed.
+#          Even if you'd like to run tests without host installation(IPMI_DO_NOT_RESTART_HOST=1), this module is still necessary as login console in this module is required.
+# Maintainer: Julie CAO <jcao@suse.com>
+
+use strict;
+use warnings;
+use base 'virt_autotest_base';
+use testapi;
+use ipmi_backend_utils;
+use Utils::Backends qw(use_ssh_serial_console is_remote_backend);
+use utils qw(zypper_call systemctl);
+use virt_autotest::utils qw(is_kvm_host is_xen_host);
+
+sub run {
+
+    #skip TW host installation and directly login
+    if (get_var('IPMI_DO_NOT_RESTART_HOST')) {
+        select_console 'sol', await_console => 0;
+        send_key 'ret';
+    }
+
+    use_ssh_serial_console unless check_screen('text-logged-in-root');
+
+    return if get_var('IPMI_DO_NOT_RESTART_HOST');
+
+    die 'Need one of both to be true: is_kvm_host || is_kvm_host' unless is_kvm_host || is_xen_host;
+    my $hypervisor = is_kvm_host ? 'kvm' : 'xen';
+    zypper_call('--gpg-auto-import-keys ref');
+    zypper_call("in -t pattern ${hypervisor}_server ${hypervisor}_tools", 1800);
+    set_serial_console_on_vh('', '', $hypervisor);
+
+    virt_autotest::utils::install_default_packages();
+
+    #libvirtd failed to be started in Tumbleweed
+    #not sure if it is a bug yet, just work it around here to make tests run
+    systemctl 'status libvirtd', ignore_failure => 1;
+    save_screenshot;
+    script_run "echo DEBUG:; echo PATH=\$PATH; which dnsmasq; which iptables; which ip6tables; which ebtables";
+    save_screenshot;
+    zypper_call('in dnsmasq iptables bridge-utils', 1800);
+    assert_script_run("ln -s /usr/sbin/$_ /sbin/$_") for qw(dnsmasq iptables ip6tables ebtables);
```

Comment:
I think:
script_run "echo DEBUG:; echo PATH=\$PATH"
for qw(dnsmasq iptables ip6tables ebtables) {
     my $path = script_output("which $_");
     assert_script_run("ln -s /usr/sbin/$_ $path");
}
is better.

Reply:
Can you @waynechen55 format your comments?

Also the `DEBUG` should not be there.
And also, Why do you create such links?

Those binaries should be executable after the installation by default.

Reply:
"code style" applied

Reply:
>And also, Why do you create such links?
It should be a bug. $PATH, libvirt or something else don't work as expected. My debug code is just a temporary workaround to facility the bug locate and fix.

Reply:
debug codes are removed as issues are located.